src/pytest_ibutsu/pytest_plugin.py

```diff
@@ -129,6 +132,27 @@ def from_config(cls, config: pytest.Config) -> IbutsuPlugin:
             enabled, ibutsu_server, ibutsu_token, ibutsu_source, ibutsu_project, extra_data, run
         )
 
+    def load_archive(self):
+        if not Path(f"{self.run.id}.tar.gz").exists():
+            return
+        cur_results = {result.test_id: result for result in self.results.values()}
+        with tarfile.open(f"{self.run.id}.tar.gz", "r:gz") as ibutsu_archive:
+            run_json = json.load(ibutsu_archive.extractfile(f"{self.run.id}/run.json"))
+            prior_run = TestRun.from_json(run_json)
+            for name in ibutsu_archive.getnames():
+                if re.match(f"{self.run.id}/{UUID_REGEX.pattern}/result.json", name):
+                    result_json = json.load(ibutsu_archive.extractfile(name))
+                    result = TestResult.from_json(result_json)
+                    if result.test_id in cur_results:
```

Comment:
@rsnyman @mshriver what do you think about this place? Should keep only the latest result of a test in the combined run or should we keep all results?

Reply:
I think relying on test_id is going to be problematic here, since the run could have multiple items with the same test_id.

That being said, this is actually a pain point in report portal use and integration, as RP has both 're-run' and 're-try' mechanisms for updating results in a run.

My preference for this would be to only keep the latest result, and to push users to create a new run to separate the test result from its previous history.